File: prime_or_not.go
//Program to find whether a given number is prime or not.package learn_go

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func check_prime(num int) bool {
	i := 2
	flag_var := true
	var remainder int
	var upper_limit int = num / 2

	for i <= upper_limit {
		remainder = num % i
		if remainder == 0 {
			flag_var = false
		}
		i += 1
	}

	return flag_var

}

func find_time(time_delta int) {
	secs := float64(time_delta / 10000000)
	hours := int(secs / 3600)
	minutes := int(secs / 60)
	seconds := int(secs) % 60

	fmt.Printf("\nThis operation took %d hour(s), %d minute(s) and %d second(s).\n", hours, minutes, seconds)

}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter the highest value till which we have to check: ")
	upper_limit_str, _ := reader.ReadString('\n')
	upper_limit_str = strings.Replace(upper_limit_str, "\r", "", -1)
	upper_limit_str = strings.Replace(upper_limit_str, "\n", "", -1)
	upper_limit, err := strconv.Atoi(upper_limit_str)

	var i int
	var prime_number []int

	if err != nil {
		fmt.Println("Error Code: ", err)
	}
	t1 := time.Now()
	for i = 3; i <= upper_limit; i += 1 {
		fmt.Println("Checking: ", i, "; timestamp: ", time.Now())
		is_prime := check_prime(i)
		if is_prime == true {
			prime_number = append(prime_number, i)
		}
	}
	t2 := time.Now()
	fmt.Printf("\nThe list of prime numbers until %d is: %v\n", upper_limit, prime_number)
	//// For Debugging; Remove for deployment
	// for j := 0; j < len(prime_number); j += 1 {
	// 	fmt.Println(prime_number[j])
	// }
	fmt.Printf("This operation took %v.\n", t2.Sub(t1))

}

File: python_src/prime_or_not.py
## Script to check for all prime numbers in a given range.
from datetime import datetime
from time import time


def check_prime(num:int) -> bool:
    flag_var:bool = True
    upper_limit = num//2
    i = 2
    while i <= upper_limit:
        if num%i == 0:
            flag_var = False
        i += 1
    
    return flag_var

def claculate_time(total_seconds:int) -> str:
    total_minutes = total_seconds//60
    seconds = total_seconds - (total_minutes*60)
    hours = total_minutes//60
    minutes = total_minutes - (hours*60)

    return f"\nThis process took {hours} hour(s), {minutes} minute(s) and {seconds} second(s)."

def main():
    limit = int(input("Enter the upper limit of the range: "))

    i: int = 3
    list_of_primes = []

    t1 = time()
    while i<=limit:
        print(f"\nChecking: {i}")
        if check_prime(i):
            list_of_primes.append(i)
        i += 1
    t2 = time()

    print(f"\nThe list of Prime Numbers until {limit} is: \n{list_of_primes}")
    print(claculate_time(t2-t1))

if __name__ == "__main__":
    main()
    
    

